packages/relay/src/lib/eth.ts

```diff
@@ -479,8 +479,16 @@ export class EthImpl implements Eth {
     if (blockNumber === 0) {
       return 0;
     } else {
-      const accountInfo = await this.sdkClient.getAccountInfo(address);
-      return Number(accountInfo.ethereumNonce);
+      const result = await this.mirrorNodeClient.resolveEntityType(address);
+      if (result && result.type === 'account') {
+        const accountInfo = await this.sdkClient.getAccountInfo(result.entity.account);
+        return Number(accountInfo.ethereumNonce);
+      }
+      else if (result && result.type === 'contract') {
+        return 1;
```

Comment:
We need to dig deeper into this one. 
Some contract addresses return `0x1` and others return `0x2`

Reply:
@shemnon do you know the implementation details of `eth_getTransactionCount` when it comes to `contract` addresses?